opentelemetry_api/opentelemetry/context/traceparent.c

```diff
@@ -0,0 +1,106 @@
+/* Copyright 2019, OpenTelemetry Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "traceparent.h"
+
+status_t _uint_from_hex_string(unsigned int* value, const char* s) {
```

Comment:
> 1. Pointers to strings without lengths are IMO problematic. If you don't want to incur a runtime cost in the release version at the minimum I would `assert` the length and !nullptr for `s`.

Length check is covered here (by checking if c is not `'\0'`).
`nullptr` check is not covered since this function is considered as **unchecked** (not public), it is intended to be called only by the OpenTelemetry C++ API/SDK code, which has already done the `nullptr` check.

> 2. However instead of `char*` I think we should be using C++ and either std::string or std::string_view (char* should be a justified exception).

`std::string` requires memory allocation if the input is a plain buffer, I was trying to avoid memory allocation here.
`std::string_view` requires very recent version of C++ 17.

Reply:
Use a back-ported string_view as I was suggesting [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/9#discussion_r337279755)

It keeps the same interface as std::string_view but doesn't require c++17. That's what we did with the opentracing-cpp [api](https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/string_view.h)

You can even a prepreocessor switch to opt into std::string_view if you're using c++17 (what absl does)